docs/DeveloperGuide.md

```diff
@@ -4,26 +4,160 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+# 1. Overview
+MeetingJio is a program for **finding potential time slots for team meetings based on everyone’s availability**.
+
+# 2. Design 
+
+## 2.1 Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 2.2 Events
+
+## 2.3 Timetables
+
+## 2.4 Commands
+
+# 3. Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## 3.1 Add Feature
+`add_lesson` allows users to create a new lesson and add it to their own timetables
+
+`add_meeting` allows users to create a new meeting and add it to all the existing timetables
+
+The following sequence diagram shows how the `add_lesson` operation generally works.
+![AddLessonSequenceDiagram](diagrams/AddLessonSequenceDiagram.png)
+
+The following sequence diagram shows how the `add_meeting` operation works in detail.
+![AddMeetingCommandSequence](diagrams/AddMeetingCommandSequence.png)
```

Comment:
Do you think that the formatting of the conditions is ambiguous?
`check if meeting clash with any event` and `if meeting exists already` could be better phrased as a boolean and `add meeting` does not tell me the condition for this block
![image](https://user-images.githubusercontent.com/19286299/160963167-db474262-189f-42c5-b3fc-3220ed399323.png)
![image](https://user-images.githubusercontent.com/19286299/160963182-2f4852af-20de-4461-90e2-e27f7d8b91c9.png)